src/lib/operators/difference.cpp

```diff
@@ -81,7 +81,7 @@ std::shared_ptr<const Table> Difference::_on_execute() {
     Segments output_segments;
 
     // creating a map to share pos_lists (see table_scan.hpp)
-    std::unordered_map<std::shared_ptr<const PosList>, std::shared_ptr<PosList>> out_pos_list_map;
+    std::unordered_map<std::shared_ptr<const AbstractPosList>, std::shared_ptr<PosList>> out_pos_list_map;
```

Comment:
Yes. We could rename it. What would be your preferred name? What about `VectorPosList`?

Reply:
A PosList that stores only chunk offsets would use a vector, too, so that might conflict.

RowIDPosList?
GenericPosList?

Maybe we'd could try to find two more names before we settle. Also pinging @Bensk1.

Reply:
I think `RowIDPosList` is the best candidate. All PosLists differ only in how they internally store the data, so naming them after what they store internally makes sense.

Other names that came to my mind are based on synonyms for "Generic": `DefaultPosList`, `IntuitivePosList`, `NaivePosList` - but I think they are worse.

Any other ideas @XPerianer @leogeier

Reply:
Yepp, makes sense. Just in case this remains the last open item for the PR: Let's go for RowIDPosList then if there is no additional input to the discussion.